### DISCUSSION TITLE
Conda-forge distribution for pygfx?

### DISCUSSION THREAD

**Discussion by Krande** [pygfx/pygfx#578] [Ideas]
Hey, I am a big fan of what you're doing with pygfx and I was just wondering if you guys have considered distributing pygfx on conda-forge in addition to pypi?

For me this would enable adding pygfx as a dependency to some of my conda packages :) 

If you want I can help setting up a conda-forge distribution of pygfx.

Let me know what you think?

Best Regards
Kristoffer

**Comment by Korijn** [comment]
I think you would have to start by checking if our dependencies are available:

https://github.com/pygfx/pygfx/blob/6ad2303e0fbf929f4f06548a624b00e20c5bc467/setup.py#L18-L25

And their dependencies (recursively) of course.

The upside is that our build process does not require any compile steps, but it does include some data files (fonts etc) and download steps of binaries (wgpu-native).

So it should be relatively easy.

We're not interested in maintaining the conda-forge recipe ourselves, but we are happy to support with any questions you may have.

**Reply by Krande** [reply]
@Korijn Thanks for responding so quickly! 

Sounds good! From the direct dependencies it looks like _wgpu_ and _pylinalg_ are not on conda-forge. I'll follow the dependency trail and add any missing ones to the same conda packaging recipe.

For early testing of the conda packages I think I'll start by publishing _pygfx_ and related packages (not on conda-forge) to my own public conda channel (channel=krande) using github actions before starting the process of setting up a more "formal" conda-forge pipeline.

**Comment by kushalkolar** [comment]
I'd probably be willing to take up maintenance of a conda-forge package for the entire pygfx stack because we will eventually need it for fastplotlib. I'm in the process of moving institutions so I'll actually have the time to do this! We maintain a few conda packages and it's not a lot of work.

**Comment by Krande** [comment]
@kushalkolar that would be great! If it can help, I did get as far as making working meta.yaml files and build script for pygfx, pylinalg and wgpu here https://github.com/Krande/condapackaging/tree/main/src/pygfx. I also published them as a test on my Krande conda channel.

**Reply by Korijn** [reply]
I created two issues with some suggestions for improvement. Looks good!

**Comment by kushalkolar** [comment]
@Krande Mark Harfouche has created conda-forge packages for both `pygfx` and `wgpu-py`:

https://anaconda.org/conda-forge/pygfx
https://anaconda.org/conda-forge/wgpu-py

Side note: we met Mark a few months ago, I guess we've sold him on the `wgpu` stack :smile: , they run a cool imaging company: https://www.ramonaoptics.com/solutions
